### DISCUSSION TITLE
Using 5 base stations in a 7x5 area

### DISCUSSION THREAD

**Discussion by sAz-G** [bitcraze/discussions#367] [Q&A]
Hi everyone,

I am trying to install 5 v2 lighthouse base stations in 7x5 room. 

The problem is that the drone does not receive signals properly from the base stations.

On [this page](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/lighthouse/multi_base_stations/) it is stated that the base stations should overlap, but one should avoid too many in one spot. I tried different positions and different directions, but the drone still does not receive a signal.

I followed the instruction mentioned in the above link (changing the configurations, build and flash). I also followed the steps mentioned in [this tutorial](https://www.bitcraze.io/documentation/tutorials/getting-started-with-lighthouse/).

Before trying to install 5 base stations, I installed 4 base stations and the installation worked fine, but not the whole flying area was covered by lighthouse signals. That makes the flight a bit unstable in the less covered area, so I decided to add a 5th base station, but then the drone have problems with receiving signals from the base stations.

Here is a screenshot of the cfclient: (the blue dot drifts away, and the shown base stations are from previous configs)
![Screenshot from 2022-11-04 15-42-45](https://user-images.githubusercontent.com/66619540/200004990-5f917665-fc21-4e4d-8ed2-44b3c145206f.png)

**Comment by krichardsson** [comment]
Hi,
this is slightly uncharted territory. I have done some brief testing with 5 base stations in the same room and I think it might be problematic, but again I'm not sure.

The problem is that the base stations interfere with each other and when more than 4 is visible at the same time it seems as they more or less block each other completely. It is unclear to me how much of an overlap that is needed to cause problems, or if it even is enough with reflected light?
You can easily verify that that the overlap is the problem by turning one base station off, which should make the system receive the other base stations. 

My suggestion is to try different setups and see what happens. Remember that the field that the base stations transmits light in is very wide, almost 180 degrees. One interesting experiment could be to block the light in some sectors to avoid unwanted overlaps in some areas.

Please share if you reach any finding, again this is an area that we have not explored.

**Comment by sAz-G** [comment]
First update:

I have tried different setups and observed that indeed, if more than 4 base stations are visible at the same time, they block each other completely. Even the slightest overlap with 6 base stations is enough to block each other completely.  I also unplugged two base station, to have only 4 overlapping, and saw that the drone could receive the signal properly.

Then I continued to try different setups with 6 base stations, until the drone could receive a signal. For this, I had to put the base stations in a setup where not more than 4 base stations are visible at the same time at the same point in the room. In other words, no matter where the drone is located, it "can see" only four base stations. See also figure below to get an idea about the setup. The setup is symmetric and there are 6 base stations in the room (2 are missing in the picture). Zoom in to see the direction of the base stations.

![20221117_191032](https://user-images.githubusercontent.com/66619540/203655444-2c35ef47-bca6-4800-b817-18e65dd7e2d5.jpg)

The above setup worked, and the drone could fly, but if I would create a new setup by turning one of the base stations on the long side even a bit towards the middle, the drone won't receive a signal and I would not be able to do the geometry estimation procedure. In order to be able to turn the base stations, I added a piece of paper on each side of the base station to block the signal. Then I could turn the base stations a bit to the middle of the room and the drone could fly (after doing geometry estimation). Below is a figure of a base station with 3 papers.

![20221117_172008](https://user-images.githubusercontent.com/66619540/203658368-3ce29f2c-dad5-40a3-bf1e-12d356ed3268.jpg)

I also used a paper with a different shape to block the signal, but this one did not work well. However, this is something that I have to check again.

I also removed possible reflectors from the room, like the light reflectors on the ceiling, to check if reflected light has a negative effect, but I could not come to a conclusion. I only observed that the flight behavior was the same and the geometry estimation was not affected by the removal of the reflectors. So I am still not sure if reflections (maybe from another source) affect the performance. 

For now, I tend to think that even a bit of overlap between the base stations causes problems, and I am not sure about reflections. 

Blocking the signal of the base stations worked well and makes it easier to use a setup with 6 base stations in the same room.

I hope I could help, and I hope to have new results soon.

**Reply by krichardsson** [reply]
I think this is very good information for anyone that wants to set up larger systems, thanks for sharing! 

> For now, I tend to think that even a bit of overlap between the base stations causes problems

Yes, if more than 4 base stations is visible at the same time, nothing works.
